redisson/src/main/java/org/redisson/cluster/ClusterConnectionManager.java

```diff
@@ -566,8 +566,8 @@ private ClusterPartition find(Collection<ClusterPartition> partitions, Integer s
                     continue;
                 }
                 masterFound = true;
-                // current master marked as failed
-                if (!newPart.isMasterFail() || newPart.getSlotsAmount() == 0) {
+                // skip the new master if it is marked as failed or has no slots
+                if (newPart.isMasterFail() || newPart.getSlotsAmount() == 0) {
```

Comment:
I'll revert this line, since if master failed, its slots should be migrated. newPart and currentPart are the same in this case.

Reply:
Can you please explain more on that? From my understanding, master change is executed before slot migration.

Also, I notice that `checkMasterNodesChange` uses address for matching while `checkSlotsMigration` uses slotId for matching. From what I have observed before (not sure if this is the case everytime), I have seen Elasticache maintenance changed slotIds but kept the addresses. For example,
```
BEFORE:
47176a5d303bd7fb11dd359130b00caf61291e4b 10.0.98.115:6379@1122 master - 0 1625764753000 7 connected 0-1092
199f14462d237348d0d530b5fb11e41602f5eab4 10.0.105.223:6379@1122 master - 0 1625764755000 14 connected 6556-7647
8f5b3b6b35eee2dfdac612a9008c154b93a03e0e 10.0.100.211:6379@1122 master - 0 1625764755163 9 connected 14200-15291
0199a2ee1fb4475b35401a89eff7d0dee64afad7 10.0.40.225:6379@1122 master - 0 1625764754062 4 connected 10924-12015
076e919bf75e67f3c87bf2eff37bfe6df32b1590 10.0.62.205:6379@1122 myself,master - 0 1625764752000 6 connected 13108-14199
88aa18acad937918c488d8c48fb13d0616dc202d 10.0.51.132:6379@1122 master - 0 1625764754000 1 connected 7648-8739
f1342e42efdf0aa19d91c37b91be7e8688814051 10.0.45.165:6379@1122 master - 0 1625764753562 0 connected 4372-5463
c155cb737c451b0baec0d6235ec31770480a54c0 10.0.172.50:6379@1122 master - 0 1625764754122 2 connected 1093-2185
516d0a51b01d609450d9ad55ba79ceaf7ef49ece 10.0.187.84:6379@1122 master - 0 1625764756000 8 connected 12016-13107
2ca9e618a98c65d56ea8f4d177736fea6606362c 10.0.169.218:6379@1122 master - 0 1625764751119 12 connected 15292-16383
b2b8a4dccd684fcdb8bc1a9a14b7d23e0ba12ae0 10.0.112.91:6379@1122 master - 0 1625764755000 5 connected 9832-10923
be614b211fa601c9e504b5fc7f7412de136828c8 10.0.60.6:6379@1122 master - 0 1625764754000 3 connected 8740-9831
a2385c4838c388b4306ef01fd10af04315a1a23d 10.0.184.88:6379@1122 master - 0 1625764756163 13 connected 2186-3278
3039cc9caa8ca7858df5b1666c20cb86f918d168 10.0.125.148:6379@1122 master - 0 1625764757124 10 connected 3279-4371
612ce4d4696525b7228ab74a05f5fd3e0d7d5d77 10.0.172.8:6379@1122 master - 0 1625764753000 11 connected 5464-6555

AFTER:
a7598423525c7a07f870897fe8cf876386b04172 10.0.98.115:6379@1122 master - 0 1626108628000 26 connected 0-1092
3e0936fb7a4f583799437544b98e770572892bcd 10.0.45.165:6379@1122 master - 0 1626108629872 27 connected 4372-5463
b52a847323d68e36729c14c5bf59f990e889a9af 10.0.125.148:6379@1122 master - 0 1626108629000 19 connected 3279-4371
fb1a209552feffe810eb1310661a29e2401e1ce3 10.0.172.8:6379@1122 master - 0 1626108628000 18 connected 5464-6555
f87132346aa23983a8777427c86aef1cdf0bf670 10.0.100.211:6379@1122 master - 0 1626108628000 30 connected 14200-15291
4681da03d00ceed968eb547730a6fbda748fdf40 10.0.187.84:6379@1122 master - 0 1626108627000 23 connected 12016-13107
6c240f43a21295889fbb763ef17ed48c84610e0d 10.0.105.223:6379@1122 master - 0 1626108630000 16 connected 6556-7647
4fbe640e2ea1d9075bdf7eeb5bdab421ccbd08a3 10.0.172.50:6379@1122 master - 0 1626108631923 31 connected 1093-2185
1aef53af6e04b3b3dad5a4ef36022dc844010865 10.0.184.88:6379@1122 master - 0 1626108628000 25 connected 2186-3278
1eadbebf70a0e95f0e00da8b3e74fac911733a46 10.0.62.205:6379@1122 myself,master - 0 1626108625000 29 connected 13108-14199
8458b5668dd1cd125bdb55c71e1a40ebfb4e3423 10.0.169.218:6379@1122 master - 0 1626108631619 21 connected 15292-16383
5fa371c5945bfbcef35ad42f73271184e0ce57fa 10.0.40.225:6379@1122 master - 0 1626108627000 28 connected 10924-12015
e8200dffd3c56aa4319b94297846995c04a312c1 10.0.112.91:6379@1122 master - 0 1626108630000 20 connected 9832-10923
080ddca7a836ba72eeb806fa55abedd71694f054 10.0.51.132:6379@1122 master - 0 1626108630917 22 connected 7648-8739
7fc94bb0ca3650872a63b5b04ac8fd72cdeca7b7 10.0.60.6:6379@1122 master - 0 1626108631000 24 connected 8740-9831
```

Reply:
> Can you please explain more on that? From my understanding, master change is executed before slot migration.

`changeMaster` method is executed only in `checkMasterNodesChange`.